src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -195,13 +243,20 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
                                                                        options.ortoolsSolver);
     auto legacyOrtoolsFiller = std::make_unique<LegacyFiller>(&Probleme);
     std::vector<LinearProblemFiller*> fillersCollection = {legacyOrtoolsFiller.get()};
-    LinearProblemData LP_Data;
-    FillContext fillCtx(0, 167);
+    std::vector<std::unique_ptr<ComponentFiller>> componentFillers;
+
+    fillModelerComponents(componentFillers, fillersCollection, problemeHebdo->modelerSystem_);
```

Comment:
About **fillModelerComponents**.
This function builds a vector of component fillers from a list of components (found in the modeler system), and then add this vector to the fillers collection.
So this function makes several things, and that's why it has more arguments than it should.

We could have instead : 
```c++
std::vector<LinearProblemFiller*> fillersCollection = {legacyOrtoolsFiller.get()};

auto componentFillers = buildComponentFillers(problemeHebdo->modelerSystem_);
std::ranges::copy(componentFillers, std::back_inserter(fillersCollection));
```
maybe more clear.

I tried it, please see this [tiny PR](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2719)

Reply:
This doesn't work because of a memory issue.
It seems that : 
```c++
std::vector<std::unique_ptr<ComponentFiller>> componentFillers;
```
is defined to keep pointed instances alive. 
Trying to shortcut this causes crashes.